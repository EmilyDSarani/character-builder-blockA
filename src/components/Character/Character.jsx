import React from 'react'

export default function Character({animal, shirt, pant, shoe}) {
    return (
        <section style={{display: 'flex', flexDirection: 'column'}}>
            <article
            className='animal'
            style={{
                backgroundImage: `url(./images/${animal}-head.png)`,
                height: 200,
                width: 200,
                backgroundSize: 'cover',
                backgroundRepeat: 'no-repeat'
            }}
            ></article>
            <article
            className='shirt'
            style={{
                backgroundImage: `url(./images/${shirt}-shirt.png)`,
                height: 200,
                width: 200,
                backgroundSize: 'cover',
                backgroundRepeat: 'no-repeat'
            }}
            ></article>
               <article
            className='pant'
            style={{
                backgroundImage: `url(./images/${pant}-pants.png)`,
                height: 200,
                width: 200,
                backgroundSize: 'cover',
                backgroundRepeat: 'no-repeat'
            }}
            ></article>       
            <article
            className='shoe'
            style={{
                backgroundImage: `url(./images/${shoe}-shoes.png)`,
                height: 200,
                width: 200,
                backgroundSize: 'cover',
                backgroundRepeat: 'no-repeat'
            }}
            ></article>
        </section>
    )
}
