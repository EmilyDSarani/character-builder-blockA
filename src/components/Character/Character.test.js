import {screen, render} from '@testing-library/react';
import Character from './Character'

it('should render the Character', () => {
    const {container} = render(<Character animal='lion' />);

    const animal = screen.getByLabelText(/animal/)

    expect(animal).toHaveStyle(`background-image: url(./images/lion-head.png)`);
    expect(container).toMatchSnapshot();
})