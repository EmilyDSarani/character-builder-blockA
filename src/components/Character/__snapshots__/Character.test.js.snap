// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Character 1`] = `
<div>
  <section
    style="display: flex; flex-direction: column;"
  >
    <article
      aria-label="animal"
      class="animal"
      style="background-image: url(./images/lion-head.png); height: 200px; width: 200px; background-size: cover; background-repeat: no-repeat;"
    />
    <article
      class="shirt"
      style="background-image: url(./images/undefined-shirt.png); height: 200px; width: 200px; background-size: cover; background-repeat: no-repeat;"
    />
    <article
      class="pant"
      style="background-image: url(./images/undefined-pants.png); height: 200px; width: 200px; background-size: cover; background-repeat: no-repeat;"
    />
    <article
      class="shoe"
      style="background-image: url(./images/undefined-shoes.png); height: 200px; width: 200px; background-size: cover; background-repeat: no-repeat;"
    />
  </section>
</div>
`;
